### DISCUSSION TITLE
SimpleTTS output from A2DP

### DISCUSSION THREAD

**Discussion by d3mac123** [pschatzmann/ESP32-A2DP#469] [Q&A] [answered]
Is it possible to have the text-to-speech output to a BT headset? This is the code I am trying to work (based on one your examples) - I tried to use the a2dp_source as audio output but it did not compile:

```
#include "BluetoothA2DPSource.h"
BluetoothA2DPSource a2dp_source;

#include "SimpleTTS.h"
NumberToText ntt;

#include "AudioCodecs/CodecMP3Helix.h"
TimeToText ttt; // Text source
I2SStream i2s;  // audio output via I2S 
MP3DecoderHelix mp3;  // mp3 decoder
AudioDictionary dictionary(ExampleAudioDictionaryValues);
TextToSpeech tts(ttt, i2s, mp3, dictionary); <-- I TRIED TO CHANGE i2s AS THE OUTPUT BY a2dp_source WITHOT SUCCESS

void setup() {
  Serial.begin(115200);
  a2dp_source.set_auto_reconnect(true);
  a2dp_source.start("Lightspee");  
  a2dp_source.set_volume(100);
  a2dp_source.write_data(data);

    // setup i2s
    auto cfg = i2s.defaultConfig(); 
    cfg.sample_rate = 24000;
    cfg.channels = 1;
    i2s.begin(cfg);

    ttt.say(14,40);
}

void loop() {

}

```

**Comment by pschatzmann** [comment, marked as answer]
The easiest might be to use the A2DPStream class as output instead of the I2SStream . However be warned: If you are new to the ESP32 or Arduino the BluetoothA2DPSource scenario is quite tricky and not the right place to start...

Here is the corresponding [simple stream example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-generator-a2dp/streams-generator-a2dp.ino)

And don't forget that the provided data format from TTS does not match with the A2DP requirement, so it needs to be converted.

ps. This is the worst place for this discussion: I think this should actually be rather in AudioTools or SimpleTTS

**Reply by d3mac123** [reply]
Just moved this to https://github.com/pschatzmann/arduino-audio-tools/discussions/987
